phonebook-api/src/main/java/com/telran/phonebookapi/service/UserService.java

```diff
@@ -0,0 +1,72 @@
+package com.telran.phonebookapi.service;
+
+import com.telran.phonebookapi.entity.ConfirmationToken;
+import com.telran.phonebookapi.entity.User;
+import com.telran.phonebookapi.exception.UserAlreadyExistsException;
+import com.telran.phonebookapi.repository.UserRepository;
+import lombok.AllArgsConstructor;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.text.MessageFormat;
+import java.util.Optional;
+
+@Service
+@AllArgsConstructor
+public class UserService {
+
+    private final UserRepository userRepository;
+    private final ConfirmationTokenService confirmationTokenService;
```

Comment:
we can use here the confirmation token repo directly. No need in ConfirmationTokenService